src/optimizer/sampling_pushdown.cpp

```diff
@@ -0,0 +1,25 @@
+#include "duckdb/optimizer/sampling_pushdown.hpp"
+#include "duckdb/planner/operator/logical_get.hpp"
+#include "duckdb/planner/operator/logical_sample.hpp"
+#include "duckdb/common/types/value.hpp"
+namespace duckdb {
+
+unique_ptr<LogicalOperator> SamplingPushdown::Optimize(unique_ptr<LogicalOperator> op) {
+	if (op->type == LogicalOperatorType::LOGICAL_SAMPLE &&
+	    op->Cast<LogicalSample>().sample_options->method == SampleMethod::SYSTEM_SAMPLE &&
+	    op->Cast<LogicalSample>().sample_options->is_percentage && !op->children.empty() &&
+	    op->children[0]->type == LogicalOperatorType::LOGICAL_GET &&
+	    op->children[0]->Cast<LogicalGet>().function.name == "seq_scan" && op->children[0]->children.empty()) {
+		auto &get = op->children[0]->Cast<LogicalGet>();
```

Comment:
Rather than hard-coding `seq_scan` here, can we add a property `supports_sample_pushdown` to the `TableFunction`, similar to how table filters work?